src/bundle.ts

```diff
@@ -61,213 +61,41 @@ export class Bundler extends Transform {
-          inlineScripts: true,
-          inlineCss: true,
-          inputUrl: fragmentPath,
-        });
-        vulcanize.process(null, (err: any, doc: string) => {
-          if (err) {
-            reject(err);
-          } else {
-            resolve({
-              url: fragmentPath,
-              contents: doc,
-            });
-          }
-        });
-      }));
-    }
-    // vulcanize the shared bundle
-    if (!this.config.shell && sharedDeps && sharedDeps.length !== 0) {
-      logger.info(`generating shared bundle...`);
-      promises.push(this._generateSharedBundle(sharedDeps));
-    }
-    const vulcanizedBundles = await Promise.all(promises);
-    const contentsMap = new Map();
-    for (const bundle of vulcanizedBundles) {
-      contentsMap.set(bundle.url, bundle.contents);
-    }
-    return contentsMap;
-  }
-
-  _addSharedImportsToShell(bundles: Map<string, string[]>): string {
-    console.assert(this.config.shell != null);
-    const shellUrl = urlFromPath(this.config.root, this.config.shell);
-    const shellUrlDir = posixPath.dirname(shellUrl);
-    const shellDeps =
-        bundles.get(shellUrl).map((d) => posixPath.relative(shellUrlDir, d));
-    logger.debug('found shell dependencies', {
-      shellUrl: shellUrl,
-      shellUrlDir: shellUrlDir,
-      shellDeps: shellDeps,
-    });
-
-    const file = this.analyzer.getFile(this.config.shell);
-    console.assert(file != null);
-    const contents = file.contents.toString();
-    const doc = parse5.parse(contents);
-    const imports = dom5.queryAll(
-        doc,
-        dom5.predicates.AND(
-            dom5.predicates.hasTagName('link'),
-            dom5.predicates.hasAttrValue('rel', 'import')));
-    logger.debug('found html import elements', {
-      imports: imports.map((el) => dom5.getAttribute(el, 'href')),
-    });
-
-    // Remove all imports that are in the shared deps list so that we prefer
-    // the ordering or shared deps. Any imports left should be independent of
-    // ordering of shared deps.
-    const shellDepsSet = new Set(shellDeps);
-    for (const _import of imports) {
-      const importHref = dom5.getAttribute(_import, 'href');
-      if (shellDepsSet.has(importHref)) {
-        logger.debug(`removing duplicate import element "${importHref}"...`);
-        dom5.remove(_import);
-      }
+      strategy = generateShellMergeStrategy(
+          urlFromPath(this.config.root, this.config.shell));
     }
-
-    // Append all shared imports to the end of <head>
-    const head = dom5.query(doc, dom5.predicates.hasTagName('head'));
-    for (const dep of shellDeps) {
-      const newImport = dom5.constructors.element('link');
-      dom5.setAttribute(newImport, 'rel', 'import');
-      dom5.setAttribute(newImport, 'href', dep);
-      dom5.append(head, newImport);
+    const bundleEntrypoints = Array.from(this.config.allFragments);
+    if (!!this.config.entrypoint) {
+      bundleEntrypoints.push(this.config.entrypoint);
```

Comment:
Removed.  There was a (terrible?) decision to start to use the term entrypoints in polymer-bundler to refer to what build and cli refer to as fragments.  This caused some confusion.  I've reverted the bit where I try to force index.html into the bundle set until we have a better handle on how to rewrite index.html's paths etc.